### PR TITLE
DOC: add examples to ctypeslib

### LINKED ISSUE #21351: DOC: Examples in docstrings – tracking issue
"Examples, more examples, and more detailed examples" is the recurrent theme in the feedback about the NumPy documentation we received via the 2020 and 2021 NumPy user surveys. 

If you come across a docstring where a function is missing an example or more/better examples are needed, please add a comment below.

**Re: the “sprintable” label** This could be a good activity for a sprint with new contributors as most of them are NumPy users. However, one doesn’t have to be a sprint participant to contribute to this initiative.

### PR DESCRIPTION
**Pull request by m-clare**
- Add examples for ctypeslib: as_array, as_ctypes
- Partially closes #21351

### CODE REVIEW

**Comment by ngoldbaum** [comment]
I wonder why we skip the doctests in this module: https://app.circleci.com/pipelines/github/numpy/numpy/29231/workflows/5e8f2bc6-49ae-4a9a-a4a5-4a956971b316/jobs/43146/parallel-runs/0/steps/0-111?invite=true#step-111-44450_26

**Comment by ngoldbaum** [comment]
Looks like the doctests for the new docstrings pass: https://app.circleci.com/pipelines/github/numpy/numpy/29231/workflows/5e8f2bc6-49ae-4a9a-a4a5-4a956971b316/jobs/43146/parallel-runs/0/steps/0-111?invite=true#step-111-39789_26

File: numpy/ctypeslib.py [hunk 1]
```diff
@@ -541,8 +561,27 @@ def as_array(obj, shape=None):
...
+        >>> c_int_array = np.ctypeslib.as_ctypes(inferred_int_array)
+        >>> c_int_array
+        <c_long_Array_3 at 0x1071fbed0>
+
+        Create ctypes object from explicit 8 bit unsigned int ``np.array`` :
+
+        >>> exp_int_array = np.array([1, 2, 3], dtype=np.uint8)
+        >>> c_int_array = np.ctypeslib.as_ctypes(exp_int_array)
+        >>> c_int_array
+        <c_ubyte_Array_3 at 0x10755a950>
```
Comment (ngoldbaum) [inline numpy/ctypeslib.py:582]: This is fine to merge as-is, but this repr isn't really helpful. Maybe print something else?

Comment (m-clare) [reply numpy/ctypeslib.py]: I added a type check and printing the array contents itself.

**Comment by ngoldbaum** [comment]
Thanks @m-clare!

**Pull request merged**